src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -7,6 +7,8 @@
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Timers;
```

Comment:
Use `System.Threading.Timer` instead. There are 2 of them in .NET, but the Threading.Timer is more preferred.

Reply:
Hi @eerhardt , I am now using `System.Threading.Timer` instead of `System.Timers`, but I'm seeing that the `System.Threading.Timer` is not always canceling, whereas `System.Timers` is. For example, in[ this run](https://dev.azure.com/dnceng/public/_build/results?buildId=869420&view=logs&j=d1af5113-e574-5a31-f7f3-02fc40ea7b26&t=b03e2d69-ce1a-5899-6643-e9b6085777b9&l=372) with `System.Threading.Timer`, cancelation of the long running model did not occur, whereas this was not the case with `System.Timers`.

In addition, I see that `System.Timers` is wrapping around `System.Threading.Timer`. Are there any specific reasons for why `System.Timers` shouldn't be used other than the other's popularity?

Reply:
`System.Threading.Timer` is lower-level, like you said `System.Timers.Timer` is built on top of `Threading.Timer`. `Timers.Timer` is part of the `System.ComponentModel` assembly, and has dependencies on things like `Component` and `ISite`, which are out-of-date .NET Framework 1.0 concepts. Basically, it is heavier-weight and has some "cruft" on it.

> but I'm seeing that the System.Threading.Timer is not always canceling

https://github.com/dotnet/machinelearning/commit/603c0ba8bbd93e21f5fe0e268577bbd6d5fb973a#diff-300431556836fdebc33d05dceb15b3d040a3b52a302967c4ebd2c6c803b9bce2R81

Is the `Timer` object getting garbage collected? I don't see you holding on to it in that commit.

Reply:
Got it, thank you for explaining. I have made it so that the `Timer` objects are not garbage collected by refencing them outside of the `Execute()`. This way, when that function is complete, the `Timer` objects will still be accessible, which will prevent them from being GCed until the entire Experiment is complete.